### DISCUSSION TITLE
如何为“衬线体”和“无衬线体”设置不同的字体？

### DISCUSSION THREAD

**Discussion by chenfeitu** [F9y4ng/GreasyFork-Scripts#362] [问答专区(Question / Answer)]
Chrome默认的字体设置就可以为“衬线体”和“无衬线体”分别设置，一个设宋体，一个设黑体。这个脚本现在可以实现吗

**Comment by chenfeitu** [comment]
![image](https://github.com/user-attachments/assets/bd4b12f1-0a3d-4b00-b348-86984431779f)

**Comment by F9y4ng** [comment]
## 浏览器字体设置的含义
浏览器的字体设置“衬线体”和“无衬线体”分别对应了CSS样式中的 `serif 字体族` 和 `sans-serif 字体族`，其含义是在CSS样式中如果没有明确指定“衬线体”和“无衬线体”的具体字体，而是使用字体家族名称 `serif, sans-serif` 时，则调用浏览器指定的“衬线体”和“无衬线体”对应的具体字体。

**Comment by EdiTurn** [comment]
我了解上述所说浏览器字体设置的含义，但是在使用过程中遇到的情况是，有些网站如维基百科，会在标题中使用衬线体，正文使用无衬线体。在使用脚本后，两者都会被覆盖为脚本所选择的字体。

**Reply by F9y4ng** [reply]
你说的这种情况，字体渲染脚本无法自动处理，没有用户会像网站设计者那样为每个不同的元素定义不同的字体。这种需求没有任何第三方通用型脚本或扩展会花时间精力去实现。

**Reply by EdiTurn** [reply]
嗯好，那我慢慢适应一下吧，谢谢。

**Reply by Wenti-D** [reply]
> 你说的这种情况，字体渲染脚本无法自动处理，没有用户会像网站设计者那样为每个不同的元素定义不同的字体。这种需求没有任何第三方通用型脚本或扩展会花时间精力去实现。

呃，事实上，以 Chrome 为例，有一个扩展「[Advanced Font Settings](https://chromewebstore.google.com/detail/advanced-font-settings/caclkomlalccbpcdllchkeecicepbmbm)」就实现了为每种文字的三种样式自定义字体。但这个扩展不能覆盖直接写在 CSS 的字体名称。

**Reply by F9y4ng** [reply]
> > 你说的这种情况，字体渲染脚本无法自动处理，没有用户会像网站设计者那样为每个不同的元素定义不同的字体。这种需求没有任何第三方通用型脚本或扩展会花时间精力去实现。
> 
> 呃，事实上，以 Chrome 为例，有一个扩展「[Advanced Font Settings](https://chromewebstore.google.com/detail/advanced-font-settings/caclkomlalccbpcdllchkeecicepbmbm)」就实现了为每种文字的三种样式自定义字体。但这个扩展不能覆盖直接写在 CSS 的字体名称。

那是扩展调用浏览器内核API直接修改的浏览器设置。原理同上所述。
